src/type-checker.cc

```diff
@@ -769,9 +769,9 @@ Result TypeChecker::OnTableFill(Type elem_type) {
   return PopAndCheck3Types(Type::I32, elem_type, Type::I32, "table.fill");
 }
 
-Result TypeChecker::OnRefFuncExpr(Index type_index) {
+Result TypeChecker::OnRefFuncExpr(Index func_index) {
   if (features_.function_references_enabled()) {
-    PushType(Type(Type::Reference, type_index));
+    PushType(Type(Type::Reference, func_index));
```

Comment:
This looks like it might be the one place we want to actually receive a type index rather than a function index.. (but as far as I now we don't).